### PR TITLE
test: negative/unknown `rpcserialversion` should throw an init error

### PR DESCRIPTION
**Pull request by brunoerg**
This PR adds test coverage for the following init errors:
https://github.com/bitcoin/bitcoin/blob/41205bf442254d17bc7885f3b2693749da714a0e/src/init.cpp#L1025-L1030

### CODE REVIEW

File: test/functional/feature_segwit.py [hunk 1]
```diff
@@ -609,6 +609,12 @@ def run_test(self):
                 assert_equal(self.nodes[1].gettransaction(txid, True)["txid"], txid)
                 assert_equal(self.nodes[1].listtransactions("*", 1, 0, True)[0]["txid"], txid)
 
+        self.log.info('Test negative and unknown rpcserialversion throw an init error')
+        self.stop_node(0)
+        self.nodes[0].assert_start_raises_init_error(["-rpcserialversion=-1"], "Error: rpcserialversion must be non-negative.")
+        self.nodes[0].assert_start_raises_init_error(["-rpcserialversion=100"], "Error: Unknown rpcserialversion requested.")
+
+
```
Comment (stickies-v) [inline test/functional/feature_segwit.py:617]: nit: PEP8 prescribes a single blank line before a method
```suggestion

```

**Review by stickies-v — COMMENTED** [review]
No strong opinion - asking more as a question: is there any risk involved for the user by providing invalid `-rpcserialversion` values? If not, is it worth it to add this functional test?

File: test/functional/feature_segwit.py (hunk 1 above)
Comment (brunoerg) [reply test/functional/feature_segwit.py:617]: Yeah, thanks!

**Comment by brunoerg** [comment]
@stickies-v, there is no risk for the user because there are these verifications, so covering them in the functional tests ensure if user types a negative or unknown value to `rpcserialversion`, he will not be harmed. See other similar PRs like: #25511, #25451, #25358.

**Comment by stickies-v** [comment]
I understand we validate user input in init.cpp, my question was more about whether or not we should test that the user input is indeed being validated. Based on a quick search, it looks like we don't do that for all parameters (e.g. "-dbcache", "-par") and I wonder if that's just because no one's gotten to it yet, or because we should only do it for critical parameters (i.e. where wrong input can be dangerous). Because it looks like for `-rpcserialversion`, it's not critical.

So I guess I'm +0 on this, don't think there's a real need to test this, but if we generally want to move towards testing all input validation, then this would make sense.

**Comment by brunoerg** [comment]
@stickies-v, I understand your point but I believe that we should cover in the tests as much as possible, even if it is not something extremely critical. If there is a validation and costs only few lines to test it, why not? And I think there are other parameters which don't have test coverage because no one got it yet, I've done this for other ones: (https://github.com/bitcoin/bitcoin/pull/25511, https://github.com/bitcoin/bitcoin/pull/25451, https://github.com/bitcoin/bitcoin/pull/25358.).

**Comment by luke-jr** [comment]
Concept ACK

**Pull request merged**